pkg/ext/task_fetcher_test.go

```diff
@@ -19,6 +21,7 @@ var (
 	adminClient      *mocks.AdminServiceClient
 	ctx              context.Context
 	taskListResponse *admin.TaskList
+	taskFilter       = filters.Filters{}
```

Comment:
Add unit tests with taskFilter data populated. 
Same here and for other places where you have added filters.
Also include test for sort by.  I see only empty filter being used in this class